src/OddJobs/Job.hs

```diff
@@ -453,8 +465,21 @@ jobMonitor = do
         liftIO $ Dir.removePathForcibly f
 
 -- | Ref: 'jobPoller'
-jobPollingSql :: TableName -> Query
-jobPollingSql tname = "update " <> tname <> " set status = ?, locked_at = ?, locked_by = ?, attempts=attempts+1 WHERE id in (select id from " <> tname <> " where (run_at<=? AND ((status in ?) OR (status = ? and locked_at<?))) ORDER BY run_at ASC LIMIT 1 FOR UPDATE) RETURNING id"
+jobPollingSql :: TableNames -> Query
+jobPollingSql tnames =
+  "update " <> tnJob tnames <>
+  "\n  set status = ?, locked_at = now(), locked_by = ?, attempts=attempts+1" <>
+  "\nwhere id in ( select id from " <> tnJob tnames <> " j_out" <>
+  "\n              where (run_at<=now() AND (status in ? OR (status = ? and locked_at < now() - ? * interval '1 second')))" <>
```

Comment:
I am strongly conditioned, by painful experience, to as much as possible do datetime evaluation either on the database client or on the database server, but not both.  Here, I've moved all the datetime logic into the query.

Reply:
I agree about using the DB as the single source of truth for current_timetamp.